### DISCUSSION TITLE
How to send all non-declared paths to a specific target (catchall principle)

### DISCUSSION THREAD

**Discussion by dorian-y** [chimurai/http-proxy-middleware#924] [Q&A]
Hi,

I try to use the proxy to send the trafic in 4 directions :
1- Send some trafic to domain 1 for several paths
2- Send some trafic to domain 2 for several paths
3- Send all trafic to domain 3, except for some explicitly excluded paths :  homepage `/` + `/blog/**`
4- Implicit (Not declarative) routing, to send the rest ( homepage `/` + `/blog/**` ) to be managed by the Nuxt Application itself

But when i configure the proxy like this,
I see that, any Path which is not declared explicitly is sent to the Nuxt app (with a 404)
But I would like those paths to be managed by my "domain 3".

Example : when I go to : "/path-3-random" (not declared in the proxy)
I would expect domain-3.com to be used, but this is my nuxt app who is used.
And because I have no internal route for this in Nuxt : I have a 404.

> Note : I want to avoid to declare explictly all the paths that my domain-3.com can manage.

Any ideas ?
Hope to be clear enough.
Thanks for any input guys.

```
proxy: {
    options: [
      // Catch Some / Path 1
      {
        target: "https://domain-1.com",
        changeOrigin: true,
        pathFilter: [
          "/path-1a/**",
          "/path-1b/**",
        ],
      },
     // Catch Some / Path 2
      {
        target: "https://domain-2.com",
        changeOrigin: true,
        pathFilter: [
          "/path-2a/**",
          "/path-2b/**",
        ],
      },      
      // Catch all Except Exclusions
      {
        target: "https://domain-3.com",
        changeOrigin: true,
        pathFilter: [
          '!/blog/**',
          '!/**'
        ],
      },
    ],
  },
```

**Comment by chimurai** [comment]
The configuration you are using is for the future 3.0 version.  (`pathFilter` is new)

Not sure exactly how different frameworks implement HPM... 

You could try to rename `pathFilter` to `context`.
Example documentation from webpack-dev-server: https://webpack.js.org/configuration/dev-server/#devserverproxy

Best is to have a look at the 2.x documentation
https://github.com/chimurai/http-proxy-middleware/tree/v2.0.6
